src/common.jl

```diff
@@ -116,3 +136,158 @@ function my_mul_mt(
     end
     return y
 end
+
+function make_dc_powerflowdata(
+    sys,
+    time_steps,
+    timestep_names,
+    power_network_matrix,
+    aux_network_matrix,
+    n_buses,
+    n_branches,
+    bus_lookup,
+    branch_lookup,
+    temp_bus_map,
+    valid_ix,
+    converged,
+    solver_data,
+)
+    branch_type = Vector{DataType}(undef, length(branch_lookup))
+    for (ix, b) in enumerate(PNM.get_ac_branches(sys))
+        branch_type[ix] = typeof(b)
+    end
+    bus_reactivepower_bounds = Vector{Vector{Float64}}(undef, n_buses)
+    timestep_map = Dict(zip([i for i in 1:time_steps], timestep_names))
+    neighbors = Vector{Set{Int}}()
+    return make_powerflowdata(
+        sys,
+        time_steps,
+        power_network_matrix,
+        aux_network_matrix,
+        n_buses,
+        n_branches,
+        bus_lookup,
+        branch_lookup,
+        temp_bus_map,
+        branch_type,
+        timestep_map,
+        valid_ix,
+        neighbors,
+        converged,
+        solver_data,
+    )
+end
+
+function make_powerflowdata(
+    sys,
+    time_steps,
+    power_network_matrix,
+    aux_network_matrix,
+    n_buses,
+    n_branches,
+    bus_lookup,
+    branch_lookup,
+    temp_bus_map,
+    branch_type,
+    timestep_map,
+    valid_ix,
+    neighbors,
+    converged,
+    solver_data,
+)
+    bus_type = Vector{PSY.ACBusTypes}(undef, n_buses)
+    bus_angles = zeros(Float64, n_buses)
+    bus_magnitude = ones(Float64, n_buses)
+
+    _initialize_bus_data!(
+        bus_type,
+        bus_angles,
+        bus_magnitude,
+        temp_bus_map,
+        bus_lookup,
+        sys,
+    )
+
+    # define injection vectors related to the first timestep
+    bus_activepower_injection = zeros(Float64, n_buses)
+    bus_reactivepower_injection = zeros(Float64, n_buses)
+    _get_injections!(
+        bus_activepower_injection,
+        bus_reactivepower_injection,
+        bus_lookup,
+        sys,
+    )
+
+    bus_activepower_withdrawals = zeros(Float64, n_buses)
+    bus_reactivepower_withdrawals = zeros(Float64, n_buses)
+    _get_withdrawals!(
+        bus_activepower_withdrawals,
+        bus_reactivepower_withdrawals,
+        bus_lookup,
+        sys,
+    )
+
+    # Shapes to reuse
+    zeros_bus_time = () -> zeros(Float64, n_buses, time_steps)
```

Comment:
There is no need to have anonymous functions do this. The compiler can't infer this until call time. Just use `zeros(Float64, n_buses, time_steps)`

Reply:
Done